blaze/expr/collections.py

```diff
@@ -418,6 +418,7 @@ def on_right(self):
         return on_right
 
     @property
+    @memoize
     def schema(self):
```

Comment:
Actually, probably more clear is to put a `__weakref__` in the `__slots__` and then just use a `WeakKeyDictionary` for the cache of the `memoize`. This is what I have done elsewhere to prevent this leak.